optaplanner-examples/src/main/java/org/optaplanner/examples/common/experimental/impl/ConsecutiveIntervalInfoImpl.java

```diff
@@ -0,0 +1,282 @@
+            }
+        } else if (!interval.getEndSplitPoint().isBefore(clusterAfterFinalIntersectedBreak.getStartSplitPoint())) {
+            // Case: interval span final break, but does not span the first break
+            // Ex:
+            //     -----------
+            //----   -----   -----
+            var previousBreakEntry =
+                    intersectedIntervalBreakMap.firstEntry();
+            var previousBreak = previousBreakEntry.getValue();
+            previousBreak.setNextCluster(intervalCluster);
+            previousBreak.setLength(
+                    differenceFunction.apply(previousBreak.getPreviousIntervalClusterEnd(), intervalCluster.getStart()));
+            intersectedIntervalBreakMap.clear();
+            clusterStartSplitPointToNextBreak
+                    .put(((IntervalClusterImpl<Interval_, Point_, Difference_>) (previousBreak
+                            .getPreviousIntervalCluster())).getStartSplitPoint(), previousBreak);
+        } else {
+            // Case: interval does not span either the first or final break
+            // Ex:
+            //     ---------
+            //----  ------   -----
+            var finalBreak =
+                    intersectedIntervalBreakMap.lastEntry().getValue();
+            finalBreak.setLength(
+                    differenceFunction.apply(finalBreak.getPreviousIntervalClusterEnd(),
+                            finalBreak.getNextIntervalClusterStart()));
+
+            var previousBreakEntry =
+                    intersectedIntervalBreakMap.firstEntry();
+            var previousBreak = previousBreakEntry.getValue();
+            previousBreak.setNextCluster(intervalCluster);
+            previousBreak.setLength(
+                    differenceFunction.apply(previousBreak.getPreviousIntervalClusterEnd(), intervalCluster.getStart()));
+
+            intersectedIntervalBreakMap.clear();
+            clusterStartSplitPointToNextBreak.put(previousBreakEntry.getKey(), previousBreak);
+            clusterStartSplitPointToNextBreak.put(intervalCluster.getStartSplitPoint(), finalBreak);
+        }
+    }
+
+    protected void removeInterval(Interval<Interval_, Point_> interval) {
+        var intervalClusterEntry =
+                clusterStartSplitPointToCluster.floorEntry(interval.getStartSplitPoint());
+        var intervalCluster = intervalClusterEntry.getValue();
+        clusterStartSplitPointToCluster.remove(intervalClusterEntry.getKey());
+        var previousBreakEntry =
+                clusterStartSplitPointToNextBreak.lowerEntry(intervalClusterEntry.getKey());
+        var nextIntervalClusterEntry =
+                clusterStartSplitPointToCluster.higherEntry(intervalClusterEntry.getKey());
+        clusterStartSplitPointToNextBreak.remove(intervalClusterEntry.getKey());
+
+        IntervalClusterImpl<Interval_, Point_, Difference_> previousIntervalCluster = null;
+        IntervalBreakImpl<Interval_, Point_, Difference_> previousBreak = null;
```

Comment:
You may get away without this type, if you do `var previousIntervalCluster = previousBreakEntry != null ? ... : null`.
Hopefully the Java compiler in my head is not misbehaving. :-)